mailbox/noise.go

```diff
@@ -298,19 +303,38 @@ type handshakeState struct {
 
 	remoteStatic    *btcec.PublicKey // nolint
 	remoteEphemeral *btcec.PublicKey // nolint
+
+	password        []byte
```

Comment:
Perhaps this should live in something more PAKE specific?

Same comment re the payloads below.